File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        seen = {key: idx for idx, key in enumerate(nums)}

        for i in range(0, len(nums)):
            complement = target - nums[i]
            if complement in seen and seen[complement] != i:
                return [i, seen[complement]]
        
        return [0, 0]
            
        

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.cs
public class Solution {
    public int[] TwoSum(int[] numbers, int target) {

        Dictionary<int, int> seen = new Dictionary<int, int>();

        for (int i = 0; i < numbers.Length; i++) {
            int comp = target - numbers[i];

            if (seen.ContainsKey(comp)) {
                return new int[] {seen[comp], i + 1};
            }

            seen.Add(numbers[i], i + 1);
        }

        return new int[] {0, 1};
        
    }
}
